Lib/test/test_ctypes/test_arrays.py

```diff
@@ -23,6 +25,21 @@ def ARRAY(*args):
 
 
 class ArrayTestCase(unittest.TestCase):
+    def test_inheritance_hierarchy(self):
+        self.assertEqual(Array.mro(), [Array, _CData, object])
+
+        self.assertEqual(PyCArrayType.__name__, "PyCArrayType")
+        self.assertEqual(type(PyCArrayType), type)
+
+    def test_type_flags(self):
+        with self.subTest(cls=Array):
+            self.assertTrue(Array.__flags__ & Py_TPFLAGS_IMMUTABLETYPE)
+            self.assertFalse(Array.__flags__ & Py_TPFLAGS_DISALLOW_INSTANTIATION)
+
+        with self.subTest(cls=PyCArrayType):
+            self.assertTrue(PyCArrayType.__flags__ & Py_TPFLAGS_IMMUTABLETYPE)
+            self.assertFalse(PyCArrayType.__flags__ & Py_TPFLAGS_DISALLOW_INSTANTIATION)
```

Comment:
Sorry for not being clearer; the `subTest` usage would be:

```suggestion
        for cls in Array, PyCArrayType:
            with self.subTest(cls=cls):
                self.assertTrue(cls.__flags__ & Py_TPFLAGS_IMMUTABLETYPE)
                self.assertFalse(cls.__flags__ & Py_TPFLAGS_DISALLOW_INSTANTIATION)
```

Ditto for the other tests.

Reply:
I got the point, updated.